File: nodejs/speech-transcribe/basic-auth.js
const https = require("https");
const FormData = require("form-data");

// basic configuration
const clientId = "[[client_id]]";
const clientSecret = "[[client_secret]]";

// create a base64 encoded api key
const apiKey = Buffer.from(`${clientId}:${clientSecret}`).toString("base64");

(async () => {
  try {
    // use the api key as a basic token
    const authorization = `Basic ${apiKey}`;

    // populate endpoint parameters
    const fs = require("fs");
    const fullPathToFile = "[[filepath]]";
    const userAgent = "Intelligent API Sample Nodejs Code";

    const form = new FormData();
    form.append("file", fs.createReadStream(fullPathToFile));

    const options = {
      hostname: "api.intelligent-api.com",
      port: 443,
      path: "/v1/speech/transcribe",
      method: "POST",
      headers: {
        Authorization: authorization,
        "User-Agent": userAgent,
        ...form.getHeaders(), // includes Content-Type with boundary
      },
    };

    const request = https.request(options, (response) => {
      console.log(`Status Code: ${response.statusCode}`);

      response.on("data", (chunk) => {
        console.log("Response Body:", chunk.toString());
      });
    });

    request.on("error", (e) => {
      console.error(`Problem with request: ${e.message}`);
    });

    form.pipe(request);
  } catch (error) {
    console.error("Error:", error);
  }
})();

File: ruby/speech-todo/basic-auth.rb
require 'base64'
require 'net/http'
require 'json'

# basic configuration
client_id = "[[client_id]]"
client_secret = "[[client_secret]]"

# create a base64 encoded api key
api_key = Base64.strict_encode64("#{client_id}:#{client_secret}")

# use the api key as a basic token
authorization = "Basic #{api_key}"

# populate endpoint parameters
file_path = "[[filepath]]"
users_date = "[[users_date]]"
users_day = "[[users_day]]"
users_time = "[[users_time]]"
user_agent = "Intelligent API Sample Ruby Code"

# build multipart form body
boundary = "----FormBoundary#{SecureRandom.hex}"
file_data = File.read(file_path, mode: "rb")
filename = File.basename(file_path)
mime_type = MIME::Types.type_for(file_path).first&.content_type || "application/octet-stream"

body = []
body << "--#{boundary}\r\n"
body << "Content-Disposition: form-data; name=\"file\"; filename=\"#{filename}\"\r\n"
body << "Content-Type: #{mime_type}\r\n"
body << "\r\n"
body << file_data
body << "\r\n--#{boundary}--\r\n"

multipart_body = body.join

# invoke the API endpoint
url = URI("https://api.intelligent-api.com/v1/speech/todo?user_date=#{users_date}&user_day=#{users_day}&user_time=#{users_time}")

http = Net::HTTP.new(url.host, url.port)
http.use_ssl = (url.scheme == "https")

request = Net::HTTP::Post.new(url)
request["Authorization"] = authorization
request["Content-Type"] = "multipart/form-data; boundary=#{boundary}"
request["User-Agent"] = user_agent
request.body = multipart_body

response = http.request(request)

puts "Response Code: #{response.code}"
puts "Response Body: #{response.body}"
